### ISSUE TITLE
[Screen Reader-Microbit Menu Controls] Control type is not defined for the controls in the Fly out of the micro bit menu items.

### ISSUE THREAD

**Issue by v-swasha** [Microsoft/pxt#2596]
**User Experience:**
Users with Visual disabilities might not able to access the controls if control type is not defined. Like in the scenario Control type is not defined for the controls in the Fly out of the micro bit menu items.

**Test Environment**
OS: RS2 Version 1703(OS Build 15063.2)
Platform: Edge.
Screen reader: Narrator

**Repro Steps**
1-Navigate to url:- https://makecode.com/
2-Navigate to the Micro bit section element and select Code control given on it.
3-Navigate to the menu controls listed near the Simulator control for eg.Arrays
4-Navigate to the controls under micro bit menu item fly out that appears in the right and verify the issue

**Actual Result**
Control type is not defined for the controls in the Fly out of the micro bit menu items.

**Expected Result**
Control type should be defined for the controls in the Fly out of the micro bit menu items.

**Note**
This issue exists for all the controls lying in the fly outs of the micro bit menu items.

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/Documents/LCA/MAS%20-%20July%201,%202017%20Release/MAS%204.1.2%20%E2%80%93%20Name,%20Role,%20Value.docx?d=w248054a65e6847719e1e242fb5025730&csf=1

**Suggested fix:**
1-   Using ARIA role to expose the control type of user interface component 
e.g. 
<div role="toolbar" tabindex="0" id="customToolbar" >  
<img src="img/btn1.gif" role="button" tabindex="-1" alt="Home" id="b1" title="Home"> 
 
2-   Using ARIA attributes to expose the element’s state  
e.g. 
<div role="button" id="customToolbar" aria-checked=”” aria-disabled=”” aria-haspopup=”” aria-expanded=”” aria-selected=””>  
 
**Reference:** 
https://www.w3.org/TR/WCAG20-TECHS/aria.html 
https://www.w3.org/TR/WCAG20-TECHS/ARIA16.html  
https://www.w3.org/TR/WCAG20-TECHS/ARIA4.html  
https://www.w3.org/TR/WCAG20-TECHS/ARIA5.html 

**Refer Attachment**
![mas 4 1 2_makecode_control type i not defined](https://user-images.githubusercontent.com/30459830/28813563-539285b4-76b7-11e7-8174-6b5ad7882505.png)

**Comment by veler** [comment]
This is part of Blocky, a third-part component.

**Comment by abchatra** [comment]
We have got exception for Blockly bugs from Julia Liuson (CVP). Closing these as Won't Fix. Following is the reasoning.

----------------------

MakeCode can be used in either Text editor or Block editor.

Blocks editor is inherently visual and is based on google blockly. Usage requires drag and drop functionality and we don’t have a equivalence in MakeCode for this w.r.t accessibility. 
We have evaluated couple of research projects which tries to find a equivalence to block based coding for accessibility needs: 
•	Project Torino (Requires students to buy physical blocks)
•	Accessible blockly (This is text based coding without syntax errors)
Both these are drastic departure from the blockly and provides a complete new experience. These are still in research and we intend not to use them in MakeCode at this point of time.

As workaround, the text editor in MakeCode is the way we see students with the need of accessibility using MakeCode.
